server/src/middleware/asset-upload.interceptor.ts

```diff
@@ -17,7 +18,10 @@ export class AssetUploadInterceptor implements NestInterceptor {
     const checksum = fromMaybeArray(req.headers[ImmichHeader.CHECKSUM]);
     const response = await this.service.getUploadAssetIdByChecksum(req.user, checksum);
     if (response) {
-      res.status(200).send(response);
+      res.status(200);
+      // Returning the response body wrapped in an Observable will cause nestjs
+      // to use it, and skip all downstream request processing.
+      return of({ status: AssetMediaStatus.DUPLICATE, id: response.id });
```

Comment:
What does the `of()` method do?

Reply:
It makes an observable out of the object.